backend/dataall/modules/s3_datasets_shares/services/share_managers/s3_utils.py

```diff
@@ -0,0 +1,92 @@
+from enum import Enum, auto
+from typing import List
+
+from dataall.modules.s3_datasets_shares.aws.kms_client import (
+    DATAALL_BUCKET_KMS_DECRYPT_SID,
+    DATAALL_BUCKET_KMS_ENCRYPT_SID,
+    DATAALL_KMS_PIVOT_ROLE_PERMISSIONS_SID,
+    DATAALL_ACCESS_POINT_KMS_DECRYPT_SID,
+    DATAALL_ACCESS_POINT_KMS_ENCRYPT_SID,
+)
+from dataall.modules.s3_datasets_shares.aws.s3_client import (
+    DATAALL_READ_ONLY_SID,
+    DATAALL_WRITE_ONLY_SID,
+    DATAALL_MODIFY_ONLY_SID,
+)
+from dataall.modules.shares_base.services.shares_enums import ShareObjectDataPermission
+
+SID_TO_ACTIONS = {
```

Comment:
It's really cool to have mapping
The only problem is that s3 actions are still duplicated here https://github.com/data-dot-all/dataall/pull/1407
So, we may have problems with share verification

Reply:
That's a great catch Sofia, indeed it seems that we have 2 places (IAM policies and S3 policies) we set the permissions and I missed the first. Fortunately the first are being used only for cross-account sharing while the later for intra account. The initial requirement was to implement WRITE only for intra-account and as such we are good to go. The only piece that is missing is to add guardrails around this rule which I will implement and push soon.